plugin/src/main/java/io/jenkins/plugins/redis/RedisFingerprintStorage.java

```diff
@@ -153,12 +141,99 @@ public void delete(@NonNull String id) throws JedisException {
      * Returns true if there's some data in the fingerprint database.
      */
     public boolean isReady() {
-        try (Jedis jedis = getJedis()) {
+        JedisPoolManager jedisPoolManager = JedisPoolManager.INSTANCE;
+        try (Jedis jedis = jedisPoolManager.getJedis()) {
             return jedis.smembers(instanceId).size() != 0;
         } catch (JedisException e) {
             LOGGER.log(Level.WARNING, "Failed to connect to Jedis", e);
             throw e;
         }
     }
 
+    private String host = RedisFingerprintStorageDescriptor.DEFAULT_HOST;
+    private int port = RedisFingerprintStorageDescriptor.DEFAULT_PORT;
+    private int database = RedisFingerprintStorageDescriptor.DEFAULT_DATABASE;
+    private boolean ssl = RedisFingerprintStorageDescriptor.DEFAULT_SSL;
+    private int connectionTimeout = RedisFingerprintStorageDescriptor.DEFAULT_CONNECTION_TIMEOUT;
+    private int socketTimeout = RedisFingerprintStorageDescriptor.DEFAULT_SOCKET_TIMEOUT;
+    private String credentialsId = RedisFingerprintStorageDescriptor.DEFAULT_CREDENTIALS_ID;
+
+    public String getHost() {
+        return host;
+    }
+
+    @DataBoundSetter
+    public void setHost(String host) {
```

Comment:
I suggest adding these setters in a builder object so you can make the main object's instance vars immutable (final)

Reply:
It would be awesome if you could explain a little more, I am not sure that I understand. This is a singleton object, whose configuration is set (using these setters) by Jelly engine that runs behind in Jenkins [[link](https://wiki.jenkins.io/display/JENKINS/Basic+guide+to+Jelly+usage+in+Jenkins)]. I am not sure how I can decouple this

Reply:
Could this be done with DataBoundConstructor instead of setters like in this example? http://codeandme.blogspot.com/2018/11/jenkins-2-builder-plugin-some-jelly.html?m=1

Reply:
Best practices from Jenkins actually suggests using setter rather than contructor [[source](https://www.jenkins.io/doc/developer/plugin-development/pipeline-integration/#constructor-vs-setters)].

Reply:
Cool, thanks for the data, lets follow the current precedent. Generally speaking though, it isn't the best pattern because mutability within objects impacts thread safety.